### ISSUE TITLE
Question: data-slot Attribute

### ISSUE THREAD

**Issue by 1letter** [plone/plone.app.mosaic#379]
It's not an Issue, only a Question. Can anyone explain what mean the `data-slots="body > header *"` Attributes in the site-layout Templates? And why is the greater than ">" Sign replaced with character entity, [look here](https://github.com/plone/plone.app.mosaic/blob/master/src/plone/app/mosaic/layouts/site/default.html#L44)? The data-tile and the data-panel attribute is explained in the plone.app.blocks docs, but not the data-slots attribute.
If i look in my rendered Mosaic Page i found this markup:
```html
<aside id="global_statusmessage" data-slots="&gt; global_statusmessage"></aside>
```
But i didn't find the Markup with the `data-slots="body > header *"` 
Thx, Jan

**Comment by datakurre** [comment]
Yes.

https://github.com/plone/plone.app.mosaic/blob/master/src/plone/app/mosaic/browser/main_template.py#L62

It's an optional syntax for site layouts to make them support any non-mosaic view depending on main_template.

In Deco, everything was supposed to be just HTML layouts and tiles. That was revolutionary and prevented Deco from being actually used. In Mosaic there's backwards compatibility support for existing main_template based ZPT views. 

But when site layouts are defined with just HTML, panels and tiles, how can the existing main_template based views work wih those layouts: When site layouts are enabled, Mosaic overrides the default main_template view, and when a HTML site layout is defined, Mosaic creates ZPT main_template from that in fly (and caches it). data-slot-attribute is used to annotate, where Mosaic should define metal macro slots in that layout. The complexity of that syntax comes from avoiding additional markup for slots on Mosaic pages, and is: data-slots="list of macro slots to wrap children of this node > list of slots to prepend the children * lists of slots to append after the children". 

My daily use of these follows the examples in example BS site layouts in Mosaic repository. When I had to implement this, I did not realise that my usage would remain so constant over time and probably less generic syntax (or even some automatic convention) would have been enough.
